LocalWorkspace is flaky

**LocalWorkspace** has been detected as a flaky test.

- **Package:** `nodejs`
- **Category:** test_flake
- **Occurrences:** 305
- **First seen:** 2026-04-24
- **Last seen:** 2026-05-04

**CI Run:** https://github.com/pulumi/pulumi/actions/runs/25328375303

**Test output:**
```
1) LocalWorkspace
runs through the stack lifecycle with multiple inline programs in parallel:
Error: Timeout of 300000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/runner/work/pulumi/pulumi/sdk/nodejs/bin/tests/automation/localWorkspace.spec.js)
at listOnTimeout (node:internal/timers:605:17)
at processTimers (node:internal/timers:541:7)



error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [Makefile:103: test_auto] Error 1
::warning Failed after 1 attempts
##[error]Process completed with exit code 1.
##[group]Run actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02
with:
name: gotestsum-test-results-115248c111c1fa5bd0744455724d478fcef84569
path: test-results/*.json
if-no-files-found: warn
compression-level: 6
overwrite: false
include-hidden-files: false
env:
PULUMI_VERSION: 3.235.0-alpha.1777909293
PULUMI_TEST_OWNER: moolumi
PULUMI_TEST_ORG: moolumi
PULUMI_ACCESS_TOKEN: ***
PULUMI_TEST_USE_SERVICE: false
PULUMI_DISABLE_AUTOMATIC_PLUGIN_ACQUISITION: true
PYTHON: python
GO_TEST_PARALLELISM: 8
GO_TEST_PKG_PARALLELISM: 2
GO_TEST_SHUFFLE: off
PULUMI_TEST_RETRIES: 0
DOTNET_CLI_TELEMETRY_OPTOUT: true
DOTNET_ROLL_FORWARD: Major
SEGMENT_DOWNLOAD_TIMEOUT_MIN: 10
OTEL_EXPORTER_OTLP_ENDPOINT: ***
PULUMI_HOME: /home/runner/work/pulumi/pulumi/home
TEST_ALL_DEPS: 
TMP: 
TEMP: 
GOPATH: 
GOCACHE: 
TEST_HASH_ID: 115248c111c1fa5bd0744455724d478fcef84569
GENERIC_VERSION: 3.235.0-alpha.1777909293
VERSION: 3.235.0-alpha.1777909293
PYPI_VERSION: 3.235.0a1777910123
DOTNET_VERSION: 3.235.0-alpha.1777909293
GORELEASER_CURRENT_TAG: v3.235.0-alpha.1777909293
PULUMI_TEST_COVERAGE_PATH: /home/runner/work/pulumi/pulumi/coverage
PULUMI_GOCOVERDIR: /tmp/tmp.2Zrk5QZoms
GOCOVERDIR: /tmp/tmp.2Zrk5QZoms
UV_CACHE_DIR: /home/runner/work/_temp/setup-uv-cache
pythonLocation: /opt/hostedtoolcache/Python/3.14.4/x64
PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.14.4/x64/lib/pkgconfig
Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.14.4/x64
Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.14.4/x64
Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.14.4/x64
LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.14.4/x64/lib
DOTNET_ROOT: /usr/share/dotnet
JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.30-7/x64
JAVA_HOME_11_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.30-7/x64
GRADLE_ACTION_ID: gradle/actions/setup-gradle
GRADLE_BUILD_ACTION_SETUP_COMPLETED: true
GRADLE_BUILD_ACTION_CACHE_RESTORED: true
DEVELOCITY_INJECTION_INIT_SCRIPT_NAME: gradle-actions.inject-develocity.init.gradle
DEVELOCITY_AUTO_INJECTION_CUSTOM_VALUE: gradle-actions
GITHUB_DEPENDENCY_GRAPH_ENABLED: false
PULUMI_GO_DEP_ROOT: /home/runner/work/pulumi
##[endgroup]
##[warning]No files were found with the provided path: test-results/*.json. No artifacts will be uploaded.
Post job cleanup.
In post-action step
Cache is read-only: will not save state for use in subsequent builds.
Generating Job Summary
Completed post-action step
Post job cleanup.
Post job cleanup.
[command]/usr/bin/git version
git version 2.53.0
Copying '/home/runner/.gitconfig' to '/home/runner/work/_temp/fe6eb35b-a4b5-40c1-8937-f1b19a4ccb55/.gitconfig'
Temporarily overriding HOME='/home/runner/work/_temp/fe6eb35b-a4b5-40c1-8937-f1b19a4ccb55' before making global git config changes
Adding repository directory to the temporary git global config as a safe directory
[command]/usr/bin/git config --global --add safe.directory /home/runner/work/pulumi/pulumi
[command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
[command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
[command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
http.https://github.com/.extraheader
[command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
[command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
[command]/usr/bin/git config --local --name-only --get-regexp ^includeIf\.gitdir:
[command]/usr/bin/git submodule foreach --recursive git config --local --show-origin --name-only --get-regexp remote.origin.url
Cleaning up orphan processes
Terminate orphan process: pid (16717) (pulumi-language-nodejs)
##[warning]Node.js 20 actions are deprecated. The following actions are running on Node.js 20 and may not work as expected: actions/checkout@34e114876b0b11c390a56381ad16ebd13914f8d5, actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093, actions/setup-dotnet@67a3573c9a986a3f9c594539f4ab511d57bb3ce9, actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5, actions/setup-java@c1e323688fd81a25caa38c78aa6df2d33d3e20d9, actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020, actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065, actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02, astral-sh/setup-uv@d4b2f3b6ecc6e67c4457f6d3e41ec42d3d0fcb86, gradle/actions/setup-gradle@d9c87d481d55275bb5441eef3fe0e46805f9ef70, jaxxstorm/action-install-gh-release@71d17cb091aa850acb2a1a4cf87258d183eb941b, oven-sh/setup-bun@735343b667d3e6f658f44d0eca948eb6282f2b76. Actions will be forced to run with Node.js 24 by default starting June 2nd, 2026. Node.js 20 will be removed from the runner on September 16th, 2026. Please check if updated versions of these actions are available that support Node.js 24. To opt into Node.js 24 now, set the FORCE_JAVASCRIPT_ACTIONS_TO_NODE24=true environment variable on the runner or in your workflow file. Once Node.js 24 becomes the default, you can temporarily opt out by setting ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION=true. For more information see: https://github.blog/changelog/2025-09-19-deprecation-of-node-20-on-github-actions-runners/

```

--- a/sdk/nodejs/Makefile
+++ b/sdk/nodejs/Makefile
@@ -100,7 +100,7 @@ unit_tests:: $(TEST_ALL_DEPS) tools_tests
 	yarn run nyc --no-clean -s mocha 'bin/tests_with_mocks/**/*.spec.js'
 
 test_auto:: $(TEST_ALL_DEPS)
-	yarn run nyc --no-clean -s mocha --timeout 300000 --parallel 'bin/tests/automation/**/*.spec.js'
+	yarn run nyc --no-clean -s mocha --timeout 450000 --parallel 'bin/tests/automation/**/*.spec.js'
 
 test_integration:: $(TEST_ALL_DEPS)
 	node 'bin/tests/runtime/closure-integration-tests.js'
@@ -150,7 +150,7 @@ TEST_GREP ?= .*
 #
 # $ make test_watch TEST_GREP="lists tag values"
 test_watch::
-	PULUMI_TEST_ORG=$(PULUMI_TEST_ORG) yarn mocha "**/*.spec.ts" --timeout 300000 --bail -j 1 --watch --watch-files "**/*.ts" --grep "$(TEST_GREP)"
+	PULUMI_TEST_ORG=$(PULUMI_TEST_ORG) yarn mocha "**/*.spec.ts" --timeout 450000 --bail -j 1 --watch --watch-files "**/*.ts" --grep "$(TEST_GREP)"
 
 dist:: GOBIN=$(or $(shell go env GOBIN),$(shell go env GOPATH)/bin)
 dist::

